tools/bazel.rc

```diff
@@ -2,6 +2,7 @@
 
 build --client_env=CC=clang
 build --copt=-DGRPC_BAZEL_BUILD
+build --copt=-DABSL_FORCE_THREAD_IDENTITY_MODE=2
```

Comment:
as users have pointed out previously, for compilation flags that are required for a successful and working build, the `bazel.rc` file is not a good place. The issue is that when you're using gRPC from your own project, you won't automatically get the options that are declared in bazel.rc (unless you manually take them and add them to own your project's bazel.rc).

So if `-DABSL_FORCE_THREAD_IDENTITY_MODE=2` is actually required (it seems that it is), it needs to be defined elsewhere (To be fair some of the other compilation options defined here should also move).

I THINK the right approach is to use "config_setting" here. https://github.com/grpc/grpc/blob/7fccdd7385386eda004d1ccf91c3aea41832b161/BUILD#L40

FTR An example where users have run into a very similar problem (it ended up being solved in a different way, as GRPC_BAZEL_BUILD is currently not really required for building grpc library) is here:
https://github.com/bazelbuild/bazel/issues/4341#issuecomment-574707231

CC @gnossen

Reply:
Thanks, Jan! I didn't think that this PR was ready for the review but it seems that I pressed the review button :)

`ABSL_FORCE_THREAD_IDENTITY_MODE` is temporary tested by the advice from b/170119606#comment3 and I don't want to add that option if possible. (Ideally I want Abseil to determine whether it should be used by some magic) But If I have to, I will add it to the config_setting. Thanks!